it/uk/gov/hmrc/agentsubscriptionfrontend/controllers/StartControllerISpec.scala

```diff
@@ -46,30 +62,90 @@ class StartControllerISpec extends BaseISpec {
+      val url = "/foo"
+      implicit val request = FakeRequest(GET, s"?continue=$url")
+
+      val result = await(controller.start(request))
+
+      status(result) shouldBe 200
+      bodyOf(result) should include("Create your Agent Services account")
+
+      sessionStoreService.currentSession.continueUrl shouldBe Some(ContinueUrl(url))
+    }
+
+    "store the absolute www.tax.service.gov.uk continue url in the session store" in {
+      val url = "http://www.tax.service.gov.uk/foo/bar?some=true"
+      implicit val request = FakeRequest(GET, s"?continue=${URLEncoder.encode(url, "UTF-8")}")
+
+      val result = await(controller.start(request))
+
+      status(result) shouldBe 200
+      bodyOf(result) should include("Create your Agent Services account")
+
+      sessionStoreService.currentSession.continueUrl shouldBe Some(ContinueUrl(url))
+    }
+
+    "store the whitelisted absolute external continue url in the session store" in {
+      val url = "http://www.foo.com/bar?some=false"
+      implicit val request = FakeRequest(GET, s"?continue=${URLEncoder.encode(url, "UTF-8")}")
+
+      val result = await(controller.start(request))
 
       status(result) shouldBe 200
       bodyOf(result) should include("Create your Agent Services account")
 
       sessionStoreService.currentSession.continueUrl shouldBe Some(ContinueUrl(url))
     }
 
+    "not store the non whitelisted absolute external continue url in the session store" in {
+      val url = "http://www.foo.org/bar?some=false"
+      implicit val request = FakeRequest(GET, s"?continue=${URLEncoder.encode(url, "UTF-8")}")
+
+      val result = await(controller.start(request))
+
+      status(result) shouldBe 200
+      bodyOf(result) should include("Create your Agent Services account")
+
+      sessionStoreService.currentSession.continueUrl shouldBe None
+    }
+
+    "not store the absolute external continue url in the session store if the url contains an invalid character" in {
+      val url = "http://www@foo.com"
+      implicit val request = FakeRequest(GET, s"?continue=${URLEncoder.encode(url, "UTF-8")}")
+
+      val result = await(controller.start(request))
+
+      status(result) shouldBe 200
+      bodyOf(result) should include("Create your Agent Services account")
+
+      sessionStoreService.currentSession.continueUrl shouldBe None
+    }
+
     "not store the continue url in the session store if the continue param is not mentioned in the url" in {
       implicit val request = FakeRequest()
 
-      val result = await(controller.start()(request))
+      val result = await(controller.start(request))
 
       status(result) shouldBe 200
       bodyOf(result) should include("Create your Agent Services account")
 
       sessionStoreService.currentSession.continueUrl shouldBe None
     }
 
-    behave like aPageWithFeedbackLinks(request => controller.start()(request))
+    behave like aPageWithFeedbackLinks(request => controller.start(request))
 
   }
 
```

Comment:
In general, there's quite a few tests for the whitelisting in this StartControllerISpec for the /start endpoint, and it feels like they're focusing a more on the whitelisting than on the /start endpoint itself.  
Perhaps the whitelisting could be unit tested separately in this comprehensive fashion, and then just have a test or two in here that simply checks whitelisting is being applied to the root and start endpoints.
Without separate tests for the whitelisting alone, I can see that there'd be pressure to add a similar set of tests for the /returnAfterGGCredsCreated endpoint for APB-1128. Perhaps it could refactored as part of APB-1128.